source/blog/2021-04-19-dbal-2.13.1-3.1.0.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "New Release: Doctrine DBAL 2.13.1 and 3.1.0 with important Forward Compatibility fix"
+authorName: Benjamin Eberlei
+authorEmail: kontakt@beberlei.de
+permalink: /2021/04/19/dbal-2.13.1-3.1.0.html
+---
+
+Last month [we released DBAL
+2.13.0](https://www.doctrine-project.org/2021/03/29/dbal-2.13.html) as an
+important push for the ecosystem towards DBAL 3 with an extensive deprecation
+layer and forwards compatibility.
+
+We made a few mistakes, given that the forward compatibility layer is quite complex.
+As such we have now released Doctrine DBAL 2.13.1 and 3.1.0 with two new APIs
+that improve the forward compatiblity.
+
+The problem lies in `Statement::execute`: 2.13.0 would return a `bool` and 3.0.0
+would return a `Result` from this method. This is a BC break that we cannot fix
+with forward compatibility.
+
+As such we introduced two new APIs on `Statement` that replace `execute`.
+When the old code was:
+
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$statement->execute();
+
+$rows = $statement->fetchAll();
+```
+
+Then the new code is now:
+
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$result = $statement->executeQuery();
+
+$rows = $result->fetchAllAssociative();
+```
+
+The DBAL 2.13 forward compatiblity layer supports both versions of all code and
+returns a Statement/Result hybrid that has all the APIs that the DBAL Statement
+had up until version 2.12. This way you can slowly move from the old to the new
+API in your code.
+
+Thank you again to [mdumoulin](https://github.com/mdumoulin) for the work on
+improving the forward compatiblity and to [Sergei](https://twitter.com/srgmrzv)
```

Comment:
You can use my full name. I think I'm not that famous as Rasmus or Fabien to be referenced by first name :)